extensions/amp-script/amp-script.md

```diff
@@ -77,63 +109,103 @@ You can also include your JavaScript inline, in a `script` tag. You must:
 
-`amp-script` supports getting and setting [`amp-state`](https://amp.dev/documentation/components/amp-bind/#initializing-state-with-amp-state) JSON via JavaScript.
+`querySelector()` is supported for simple selectors - element, ID, class, and attribute. So, `document.querySelector('.class')` will work, but `document.querySelector('.class1 .class2')` will not. [See the code](https://github.com/ampproject/worker-dom/blob/main/src/worker-thread/dom/Element.ts#L159) for details.
 
-This enables advanced interactions between `amp-script` and other AMP elements on the page via `amp-bind` [bindings](https://amp.dev/documentation/components/amp-bind/#bindings). Invoking `AMP.setState()` from `amp-script` may cause mutations to the DOM as long as it was triggered by user gesture, otherwise it will only implicitly set state (similar to [`amp-state` initialization](https://amp.dev/documentation/examples/components/amp-bind/?referrer=ampbyexample.com#initializing-state)).
+`amp-script` supports common Web APIs like `Fetch`, `Websockets`, `localStorage`, `sessionStorage`, and `Canvas`. Presently, the `History` API is not implemented, and neither are cookies.
+
+`amp-script` does not support the entire DOM API or Web API, as this would make `amp-script`'s own JavaScript too large and slow. If there's an API you'd like to see supported, please [file an issue](https://github.com/ampproject/amphtml/issues/new) or [suggest and contribute the change yourself](https://github.com/ampproject/amphtml/blob/master/CONTRIBUTING.md). For a complete list of supported DOM APIs, see the [API compatibility table](https://github.com/ampproject/worker-dom/blob/master/web_compat_table.md).
 
 [tip type="default"]
-`AMP.setState()` requires the [`amp-bind`](https://amp.dev/documentation/components/amp-bind) extension script to be included in the document head.
+For a set of samples showing `amp-script` in use, [see here](/documentation/examples/components/amp-script/).
 [/tip]
 
+### Frameworks and libraries
+
+Presently, libraries like [jQuery](https://jquery.com) will not work with `amp-script` without modification, as they use unsupported DOM APIs. However, [@ampproject/worker-dom](https://github.com/ampproject/worker-dom/) was designed to support the APIs that popular JavaScript frameworks use. `amp-script` has been tested with frameworks like [React](http://reactjs.org/), [Preact](https://preactjs.com/), [Vue](https://vuejs.org/), [Angular](https://angularjs.org/), [Aurelia](https://aurelia.io/), and [lit-html](https://lit-html.polymer-project.org/). To keep bundle sizes small, we recommend using Preact.
+
+### Creating AMP elements
+
+You can create an AMP element dynamically using an DOM API like `document.createElement()`. However, presently only `amp-img` and `amp-layout` are supported. If you need to create a different AMP element, please upvote on [&#35;25344](https://github.com/ampproject/amphtml/issues/25344) and add a comment describing your use case.
```

Comment:
The first sentence feels misleading. I would update to something like:

The amp-script component can add any HTML element to the page DOM, but is limited to `amp-layout` and `amp-img` for AMP components.

Reply:
This is one of the bits I inherited and did not change. I agree with you :) I've written yet a third version of this... wdyt?

(I've also updated the language in the section with the relevant error below)